chain/block.go

```diff
@@ -290,6 +291,23 @@ func (b *StatelessBlock) initializeBuilt(
 			b.containsWarp = true
 		}
 	}
+
+	// transaction hash generation
+	merkleItems := make([][]byte, 0, len(b.Txs)+len(b.results))
+	for _, tx := range b.Txs {
+		merkleItems = append(merkleItems, tx.Bytes())
```

Comment:
Would it make sense to use `txID` as key and then put something useful in the value?

Reply:
will the type of transactions have distinct id for different actors? nor there will be collision